pkg/features/kube_features.go

```diff
@@ -291,6 +291,12 @@ const (
 	// deletion ordering.
 	HonorPVReclaimPolicy featuregate.Feature = "HonorPVReclaimPolicy"
 
+	// owner: @danwinship
+	// kep: https://kep.k8s.io/3015
+	//
+	// Enables PreferSameZone and PreferSameNode values for trafficDistribution
+	ImprovedTrafficDistribution featuregate.Feature = "ImprovedTrafficDistribution"
```

Comment:
This is the feature gate name I put in the KEP but having actually written all the code now, it sounds like it means "makes the distribution of traffic better", not "makes TrafficDistribution better", so I could rename it...

Reply:
It does sound like the distribution of traffic is better.
May be `ExpandedTrafficDistribution` or `ExtendedTrafficDistribution` is better, but that doesn't leave much for a next step.

`PerZoneAndPerNodeTrafficDistribution` is a long.

Reply:
`ExpandedTrafficDistribution` sounds good to me too. A future option can be something more specific -- the reason why we can't be more specific is because we're trying to account for both PreferSameNode and PreferSameZone right.

Reply:
I think "PreferSameTopologyTrafficDistribution" or "PreferSameTrafficDistribution" would work.  "Improved" is a little misleading.  Expanded is not much better.  But this doesn't matter a whole lot.